src/validation.cpp

```diff
@@ -2468,6 +2474,18 @@ bool CChainState::DisconnectTip(BlockValidationState& state, DisconnectedBlockTr
         assert(flushed);
     }
     LogPrint(BCLog::BENCH, "- Disconnect block: %.2fms\n", (GetTimeMicros() - nStart) * MILLI);
+
+    {
+        // Prune blockers that began at or after the tip should get moved backward so they get a chance to reorg
+        const int max_height_first = pindexDelete->nHeight - 1;
+        for (auto& prune_blocker : m_blockman.m_prune_blockers) {
+            if (prune_blocker.second.m_height_first <= max_height_first) continue;
+            // NOTE: Don't need to write to db here, since it will get synced when the rest of the chainstate
```

Comment:
b44fea2c This comment seems to be unfinished or missing context ("when"...what?), and not sure if "when" should be "with"

Reply:
Done